src/components/utils/src/threads/thread_posix.cc

```diff
@@ -275,10 +278,13 @@ void Thread::join() {
 }
 
 Thread::~Thread() {
-  finalized_ = true;
-  stopped_ = true;
-  join();
-  pthread_join(handle_, NULL);
+  if (!finalized_) {
+    ThreadCounter::Decrement();
+    finalized_ = true;
+    stopped_ = true;
+    join();
+    pthread_join(handle_, NULL);
```

Comment:
@OHerasym, Isn't it better to check here thread_options_.is_joinable() before calling pthread_join(handle_, NULL);?